src/declaration.c

```diff
@@ -51,6 +51,8 @@ bool checkFrameAccess(Loc loc, Scope *sc, AggregateDeclaration *ad, size_t iStar
             {
                 if (!fd->isThis() && !fd->isNested())
                     break;
+                if (FuncLiteralDeclaration *fld = fd->isFuncLiteralDeclaration())
+                    fld->tok = TOKdelegate;
```

Comment:
`checkFrameAccess` is supposed to be a pure function, merely checking things. But here it is now modifying global state. There's got to be a better way. I do note that `checkNestedReference` does it, too, so there's an excuse for it. I hope that when we switch to ddmd, we can migrate to a better design.